### PR TITLE
#43: added support for setuptools_scm

### PR DESCRIPTION
**Pull request by ptisserand**
__version__ is also set by setuptools_scm.

### CODE REVIEW

File: financeager/__init__.py [hunk 1]
```diff
@@ -3,7 +3,11 @@
 from logging import getLogger, StreamHandler, DEBUG, INFO, Formatter, handlers
 
 # versioning information
-__version__ = "0.21"
+from pkg_resources import get_distribution, DistributionNotFound
+try:
+    __version__ = get_distribution(__name__).version
+except DistributionNotFound:
+    __version__ = "0.22"
```
Comment (pylipp) [inline financeager/__init__.py]: Why did you add this as fallback?

Comment (ptisserand) [reply financeager/__init__.py]: Do you need 0.22 as fallback version ?
I shortcut output from python setup.py --version but you are right, I'm worng it should be 0.21
Should I provide a new pull request to correct default number version ?
Alos if I provide a new pull rquest, should I squash these commits ?

Comment (pylipp) [reply financeager/__init__.py]: I'd like to avoid having to manually update version information in the source code, so I'd omit handling DistributionNotFound (the version info does get lost when the code is torn out of the context of distribution using pip/setup.py or git, e.g. if anyone was to copy it around to different machines; but hey it's 2019 and package distribution is supposed to be done using pip, and package development is supposed to be done using git ;))

You could amend your commit to add the correction (do the change, `git add` it, and run `git commit --amend`. Then you have to force-push to overwrite the commit on the remote). This means you don't have to open another PR, and the current one still holds only one commit.

Comment (ptisserand) [reply financeager/__init__.py]: Ok it's clear, thanks

File: setup.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
 setup(
     name="financeager",
     version=__version__,
```
Comment (pylipp) [inline setup.py]: I think this can be omitted then, including the corresponding import.

Comment (ptisserand) [reply setup.py]: Ok, it's done

**Comment by pylipp** [comment]
Do you have the development environment set up locally? The Travis build fails due to a formatting error (maybe a superfluous newline or a whitespace). You can easily find out by running the style-checks locally.

**Comment by ptisserand** [comment]
Really sorry I didn't see installation of development environment.
It should be OK now.

**Comment by pylipp** [comment]
Thanks :)
